src/NUnitTestAdapter/NUnitEngine/NUnitDiscoveryTestClasses.cs

```diff
@@ -0,0 +1,365 @@
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Xml;
+using System.Xml.Linq;
+
+// ReSharper disable InconsistentNaming
+
+namespace NUnit.VisualStudio.TestAdapter.NUnitEngine
+{
+
+    /// <summary>
+    /// These classes are those found during the discovery in the execution phase.
+    /// </summary>
+
+    public class NUnitDiscoverySuiteBase
+    {
+        public string Id { get; }
+        public string Name { get; }
+        public string FullName { get; }
+        public int TestCaseCount { get; }
+
+        public NUnitDiscoverySuiteBase(string id, string name, string fullname, int count)
+        {
+            Id = id;
+            Name = name;
+            FullName = fullname;
+            TestCaseCount = count;
+        }
+
+        public NUnitDiscoverySuiteBase(NUnitDiscoverySuiteBase other) : this(other.Id, other.Name, other.FullName,
+            other.TestCaseCount)
+        {
+
+        }
+    }
+
+    public class NUnitDiscoverySuiteBaseProperties : NUnitDiscoverySuiteBase
+    {
+        public NUnitEventTestCase.eRunState Runstate { get; set; }
+        public virtual NUnitDiscoverySuiteBaseProperties Parent { get; set; }
+
+        public NUnitTestDiscoveryProperties NUnitTestDiscoveryProperties { get; } = new NUnitTestDiscoveryProperties();
+
+        public NUnitDiscoverySuiteBaseProperties(string id, string name, string fullname, int count, NUnitEventTestCase.eRunState runstate)
+        : base(id, name, fullname, count)
+        {
+            Runstate = runstate;
+        }
+
+        public NUnitDiscoverySuiteBaseProperties(NUnitDiscoverySuiteBaseProperties theBase) : base(theBase)
+        {
+            Runstate = theBase.Runstate;
+            foreach (var prop in theBase.NUnitTestDiscoveryProperties.Properties)
+            {
+                NUnitTestDiscoveryProperties.Add(prop);
+            }
+        }
+    }
+
+    public class DiscoveryConverter
+    {
+#pragma warning disable SA1303 // Const field names should begin with upper-case letter
+        private const string id = nameof(id);
+        private const string type = nameof(type);
+        private const string name = nameof(name);
+        private const string fullname = nameof(fullname);
+        private const string runstate = nameof(runstate);
+        private const string testcasecount = nameof(testcasecount);
+        private const string classname = nameof(classname);
```

Comment:
I asked about this in the Gitter core chat, and @ChrisMaddock pointed me to the NUnit classes.  They are not real Poco classes, and the adapter cant have a dependency to NUnit itself, since it can run on any version.  I am not aware of anyone existing in the engine itself.  I also need to add methods to them, they could be extension methods, but then the existing framework methods would hang on there, which doesnt make sense.   So, considering these things, I found it easier to just create those classes.  Seems like maintaining them will not be much effort, that interface is not really something that changes often.  If we could have a separate package representing that interface, I would be fine.

Reply:
nameof fixed.